### DISCUSSION TITLE
M5Stack Core2 touchscreen not operational in 14.1.0 lvgl-haspmota

### DISCUSSION THREAD

**Discussion by Grumium** [arendst/Tasmota#21580] [Support request] [answered]
## PROBLEM DESCRIPTION

Hi, I am missing touchscreen support on my `M5Stack Core2`. I ran the `auto-configuration` for this MCU and later tried the `haspmota_demo.tapp`. 
No matter what I try, I cannot get the touchscreen to respond.
In addition, the temperature and MPU sensor data are not showing up on the display, but in the WebUI everything seems fine.
The clock is updating on the display.
Any ideas?

## REQUESTED INFORMATION

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [x] Device used (e.g., Sonoff Basic): M5Stack Core2
- [x] Tasmota binary firmware version number used: Tasmota 14.1.0
   -  [x] Pre-compiled (Tasmota32 LVGL)
  -  [x] Self-compiled (Generic ESP32 with LVGL support enabled)
- [x] Flashing tools used: Web installer

## Expected Behavior
- The auto-configuration is supposed to work.
- The basic haspmota example is support to work including touchscreen support and updating data in the widgets.

## Additional Context

- WebUi:
![grafik](https://github.com/arendst/Tasmota/assets/47479352/b3c836e6-33b5-44fc-b28c-3ffe6a1a892e)
- Display:
![grafik](https://github.com/arendst/Tasmota/assets/47479352/27a64580-46b2-4aad-92ac-653de2a120df)

- Is this possibly related to  #21563 and/or #21146 ?
- I can exclude a hardware fault, since I have two identical M5Stack Core2 and both behave exactly the same.
- I tried to self-compile the newest firmware using tasmo-compiler. It does not make any difference.
- To my understanding the display.ini-file for the `FT6336U` should be correct.
- The display.ini file has the following content:
```
:H,ILI9342,320,240,16,SPI,1,5,18,23,15,-1,-1,38,40
:S,2,1,3,0,100,100
:B,60,0
:I
EF,3,03,80,02
CF,3,00,C1,30
ED,4,64,03,12,81
E8,3,85,00,78
CB,5,39,2C,00,34,02
F7,1,20
EA,2,00,00
C0,1,23
C1,1,10
C5,2,3e,28
C7,1,86
36,1,48
37,1,00
3A,1,55
B1,2,00,18
B6,3,08,82,27
F2,1,00
26,1,01
E0,0F,0F,31,2B,0C,0E,08,4E,F1,37,07,10,03,0E,09,00
E1,0F,00,0E,14,03,11,07,31,C1,48,08,0F,0C,31,36,0F
21,80
11,80
29,80
:o,28
:O,29
:A,2A,2B,2C,16
:R,36
:0,08,00,00,00
:1,A8,00,00,01
:2,C8,00,00,02
:3,68,00,00,03
:i,21,20
:UTI,FT6336U,I1,38,*,*
RD A0
CP 02
RTF
RT
:UTT
RDM 00 16
MV 2 1
RT
:UTX
MV 3 2
RT
:UTY
MV 5 2
RT
#
```

**Comment by Jason2866** [comment, marked as answer]
Erased my core2 fully with esptool.py. After flashing the lvgl firmware and doing autoconfig and loading after restart the `haspmota_demo.tapp` the screen and touch does work.
Mpu and temperature is not shown correctly since the Berry tapp is not updated. But moving through the screens with the touch symbols does work

**Reply by Grumium** [reply]
Thank you! I can confirm that the touchscreen works via the esptool-route on
**Tasmota 14.1.0.1 (8479da0-lvgl-haspmota)**

I ran
`esptool.py -b 115200 --port /dev/ttyACM0 erase_flash`
followed by
`esptool.py -b 115200 --port /dev/ttyACM0 write_flash 0x0 tasmota32-lvgl.factory.bin`

- Why is a flash erase through the Webinstaller not sufficient?

**Comment by Jason2866** [comment]
> Why is a flash erase through the Webinstaller not sufficient?

It is working, sometimes something fails.
